cl/search/api_utils.py

```diff
@@ -247,12 +286,211 @@ def __getitem__(self, item):
                 return []
 
     def append(self, p_object):
-        """Lightly override the append method so we get items duplicated in
+        """Lightly override the append method, so we get items duplicated in
         our cache.
         """
         self._item_cache.append(p_object)
 
 
+class CursorESList:
+    """Handles the execution and postprocessing of Elasticsearch queries, as
+    well as the pagination logic for cursor-based pagination.
+    """
+
+    def __init__(
+        self,
+        main_query,
+        child_docs_query,
+        page_size,
+        search_after,
+        clean_data,
+        version="v3",
+    ):
+        self.main_query = main_query
+        self.child_docs_query = child_docs_query
+        self.page_size = page_size
+        self.search_after = search_after
+        self.clean_data = clean_data
+        self.version = version
+        self._item_cache = []
+        self.results = None
+        self.reverse = None
+        self.cursor = None
+        self.cardinality_query = {
+            SEARCH_TYPES.RECAP: ("docket_id", DocketDocument),
+            SEARCH_TYPES.DOCKETS: ("docket_id", DocketDocument),
+            SEARCH_TYPES.RECAP_DOCUMENT: ("id", DocketDocument),
+        }
```

Comment:
This variable seems more suited as a class variable. Unless each object needs its own unique copy, I think we should make it static for efficiency 🤔

Reply:
You right. I've moved this as a class variable.